seacrowd/sea_datasets/belebele/belebele.py

```diff
@@ -0,0 +1,164 @@
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+    BUILDER_CONFIGS = [config_constructor(lang, "source", _SOURCE_VERSION) for lang in _NAMES]
+    BUILDER_CONFIGS.extend((config_constructor(lang, "seacrowd_qa", _SEACROWD_VERSION) for lang in _NAMES))
+    DEFAULT_CONFIG_NAME = "belebele_acm_Arab_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "link": datasets.Value("string"),
+                    "question_number": datasets.Value("int64"),
+                    "flores_passage": datasets.Value("string"),
+                    "question": datasets.Value("string"),
+                    "mc_answer1": datasets.Value("string"), 
+                    "mc_answer2": datasets.Value("string"), 
+                    "mc_answer3": datasets.Value("string"), 
+                    "mc_answer4": datasets.Value("string"), 
+                    "correct_answer_num": datasets.Value("string"), 
+                    "dialect": datasets.Value("string"), 
+                    "ds": datasets.Value("string"), # timedate
+                }
+            ) 
+        elif self.config.schema == "seacrowd_qa":
+            features = schemas.qa_features
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+        comps = self.config.name.split("_")
+        lang = comps[1]+"_"+comps[2]
+        path = dl_manager.download_and_extract(_URLS[_DATASETNAME])
+        file = "{path}/Belebele/{lang}.jsonl".format(path=path, lang=lang)
+
+        return datasets.SplitGenerator(
+            name=datasets.Split.TRAIN,
+            gen_kwargs={
+                "file": file,
+            },
+        )
+
+    def _generate_examples(self, file: str) -> Tuple[int, Dict]:
+        """Yields examples as (key, example) tuples."""
+        if self.config.schema == "source":
+            with open(file, "r", encoding="utf-8") as f: 
+                for key, line in enumerate(f):
+                    line = json.loads(line)
+                    yield key, line
+        elif self.config.schema == "seacrowd_qa":
+            with open(file, "r", encoding="utf-8") as f: 
+                for key, line in enumerate(f):
+                    line = json.loads(line)
+                    choices = [line['mc_answer1'], line['mc_answer2'], line['mc_answer3'], line['mc_answer4']]
+                    answer = choices[int(line['correct_answer_num'])-1]
+                    yield key, {
+                        "id": key,
+                        "question_id": line['question_number'],
```

Comment:
For the `question_id` field, as shown in the [QA schema](https://github.com/SEACrowd/seacrowd-datahub/blob/master/seacrowd/utils/schemas/qa.py#L6-L17), we expect the type to be a string instead of an integer. 

Perhaps we can simply use `str(line['question_number'])` for this one.

Reply:
Sounds good. Made the updates.